lib/GuestManager.php

```diff
@@ -97,11 +97,12 @@ public function isGuest($user = null) {
 	}
 
 	public function createGuest(IUser $createdBy, $userId, $email, $displayName = '', $language = '') {
-		$passwordEvent = new Event(null, ['password' => $this->secureRandom->generate(20)]);
+		$password = $this->secureRandom->generate(20);
+		$passwordEvent = new Event(null, ['password' => $password]);
 		$this->eventDispatcher->dispatch('OCP\PasswordPolicy::generate', $passwordEvent);
 		$this->userBackend->createUser(
 			$userId,
-			$passwordEvent->getArgument('password')
+			$password
```

Comment:
wait. I think the way I changed the code here is wrong. @juliushaertl does this magic rely on the event getting updated? Unfortunately the `getArgument` doesn't exist, hence this now does :boom: on 18

Reply:
It was available when Event extended GenericEvent in 17, but now as it a [subclass of SymfonyEvent](https://github.com/nextcloud/server/pull/17085) it doesn't work anymore with 18. 

I guess we need a dedicated event class in the password policy app for this.

Reply:
Was the move to the new events a breaking change anyway? Then we could still just change that for 18.

Reply:
Hmm so this is broken in any case, not just in regards to the vanished method. https://github.com/nextcloud/password_policy/blob/876f5c500203b3807e912d0711741185e8d75974/lib/AppInfo/Application.php#L45-L51 expects the old `GenericEvent`

> I guess we need a dedicated event class in the password policy app for this.

Hard to do if we want to make existing and new code work. Is this event used often?

Reply:
> Hmm so this is broken in any case, not just in regards to the vanished method. nextcloud/password_policy:lib/AppInfo/Application.php@876f5c5#L45-L51 expects the old GenericEvent

> Hard to do if we want to make existing and new code work. Is this event used often?

Password policy is shipped so we have dedicated branches per server relase anyway and the new guests release will be 18 only for the sidebar adjustments, so no need to keep this backward-compatible I'd say.

Otherwise we could keep the existing guests implementation with  https://github.com/nextcloud/server/pull/17834 and moving password policy to the new GenericEvent in public namespace.

Reply:
Let me just add a new event :)

Reply:
https://github.com/nextcloud/server/pull/18019 https://github.com/nextcloud/password_policy/pull/90 #209